src/Microsoft.ML.FastTree/Utils/VectorUtils.cs

```diff
@@ -119,81 +98,53 @@ public static unsafe double GetDotProduct(float[] vector1, float[] vector2, int
             return product;
         }
 
-        public static unsafe double GetMean(double[] vector)
+        public static double GetMean(double[] vector)
         {
             double sum = 0;
-            int length = vector.Length;
-            unsafe
+            for (int i = 0; i < vector.Length; i++)
             {
-                fixed (double* pVector = vector)
-                {
-                    for (int i = 0; i < length; i++)
-                    {
-                        sum += pVector[i];
-                    }
-                }
+                sum += vector[i];
             }
-            return sum / length;
+            return sum / vector.Length;
         }
 
-        public static unsafe double GetMean(float[] vector)
+        public static double GetMean(float[] vector)
         {
             double sum = 0;
-            int length = vector.Length;
-            unsafe
+            for (int i = 0; i < vector.Length; i++)
             {
-                fixed (float* pVector = vector)
-                {
-                    for (int i = 0; i < length; i++)
-                    {
-                        sum += pVector[i];
-                    }
-                }
+                sum += vector[i];
             }
-            return sum / length;
+            return sum / vector.Length;
         }
 
         public static double GetStandardDeviation(double[] vector)
         {
             return GetStandardDeviation(vector, GetMean(vector));
         }
 
-        public static unsafe double GetStandardDeviation(double[] vector, double mean)
+        public static double GetStandardDeviation(double[] vector, double mean)
         {
             double sum = 0;
-            int length = vector.Length;
             double tmp;
-            unsafe
+            for (int i = 0; i < vector.Length; i++)
             {
-                fixed (double* pVector = vector)
-                {
-                    for (int i = 0; i < length; i++)
-                    {
-                        tmp = pVector[i] - mean;
-                        sum += tmp * tmp;
-                    }
-                }
+                tmp = vector[i] - mean;
+                sum += tmp * tmp;
```

Comment:
>tmp * tmp; [](http://example.com/codeflow?start=23&length=10)

I assume we don't care about the overflow here, I know there is no change here from the old code behavior.

Reply:
Right. This will end up being `double.Infinity` if there's overflow.

You saw that I sprinkled some overflow checks throughout the code, but they're only in cases where we calculate array indicex. I didn't add any overflow checks elsewhere, including in places where we calculate array values (as opposed to indices).